plugin/api.thrift

```diff
@@ -0,0 +1,321 @@
+/**
+ * Package api defines the core interface which plugins use to talk to
+ * ThriftRw.
+ */
+
+/**
+ * Version is the plugin API version specified in this IDL. This MUST be
+ * provided in HandshakeResponses to specify which version of the API the
+ * plugin implements.
+ */
+const i32 VERSION = 1
+
+/**
+ * ServiceID is an arbitrary unique identifier to reference the diffeerent
+ * services in this request.
+ */
+typedef i32 ServiceID
+
+/**
+ * ModuleID is an arbitrary unique identifier to reference the different
+ * modules in this request.
+ */
+typedef i32 ModuleID
+
+/**
+ * TypeReference is a reference to a user-defined type.
+ */
+struct TypeReference {
```

Comment:
Do type references need more information about whether the type they point to is a primitive or not?